### PR TITLE
MAINT: rename custom inner classes

### PR DESCRIPTION
**Pull request by kohr-h**
Okay, last one, then sleep.

Some fixes to doc and other things in weighting, nothing spectacular.

### CODE REVIEW

File: odl/space/npy_ntuples.py [hunk 1]
```diff
@@ -1800,12 +1780,55 @@ def __init__(self, array, exponent=2.0, dist_using_inner=False):
...
 
             This option can only be used if ``exponent`` is 2.0.
+
+        Notes
+        -----
+        - For exponent 2.0, a new weighted inner product with array
+          :math:`w` is defined as
+
+          .. math::
+
```
Comment (adler-j) [inline odl/space/npy_ntuples.py]: Redundant blank line after .. math::

File: odl/space/npy_ntuples.py [hunk 2]
```diff
@@ -1800,12 +1780,55 @@ def __init__(self, array, exponent=2.0, dist_using_inner=False):
...
+          where :math:`b^{\mathrm{H}}` stands for transposed complex
+          conjugate and :math:`w \cdot a` for entry-wise multiplication.
+
+        - For other exponents, only norm and dist are defined. In the
+          case of exponent :math:`\\infty`, the weighted norm is
+
+          .. math::
+
+              \\lVert a\\rVert_{w, \\infty} :=
```
Comment (adler-j) [inline odl/space/npy_ntuples.py]: Why \\lVert instead of \\|?

File: odl/space/pspace.py [hunk 3]
```diff
@@ -983,6 +961,43 @@ def __init__(self, array, exponent=2.0, dist_using_inner=False):
...
+
+        - Note that this definition does **not** fulfill the limit property
+          in :math:`p`, i.e.,
+
+          .. math::
+
+              \|x\|_{w,p} \\not\\to \|x\|_{w,\infty}
+              \quad\\text{for } p \\to \infty
+
+          unless :math:`w = (1,...,1)`.
```
Comment (adler-j) [inline odl/space/pspace.py]: Perhaps a word here on why we do this

File: odl/space/npy_ntuples.py (hunk 2 above)
Comment (kohr-h) [inline odl/space/npy_ntuples.py]: Oh, written ages ago. I'll change it.

**Comment by kohr-h** [comment]
Fixed comments and rebased, merge after CI.

**Pull request merged**